clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3056,6 +3056,45 @@ bool Sema::checkTargetVersionAttr(SourceLocation LiteralLoc, Decl *D,
   enum SecondParam { None };
   enum ThirdParam { Target, TargetClones, TargetVersion };
   llvm::SmallVector<StringRef, 8> Features;
+  if (Context.getTargetInfo().getTriple().isRISCV()) {
+
+    llvm::SmallVector<StringRef, 8> AttrStrs;
+    AttrStr.split(AttrStrs, ';');
+
+    bool IsPriority = false;
+    bool IsDefault = false;
+    for (auto &AttrStr : AttrStrs) {
+      // Only support arch=+ext,... syntax.
```

Comment:
What if someone writes `arch=rv64gc;default;` do we error for that?

Reply:
The `target_version("arch=rv64gc;default;")` attribute will raise an error with the message: `"unsupported '' in the 'target_version' attribute string; 'target_version' attribute ignored"`. This occurs due to an empty `AttrStr` appearing in the attribute.

However, `target_version("arch=rv64gc;default")` will not raise an error and will now treat the version as default. To address this situation, I have added a `HasArch` guard.

Reply:
`arch=rv64gc;default` was the case I was concerned about. I didn't mean to have a semicolon at the end.

Reply:
The line `arch=rv64gc` will trigger a warning because it currently doesn't support the full architecture.

`unsupported 'arch=rv64gcv' in the 'target_version' attribute string; 'target_version' attribute ignored`

https://github.com/llvm/llvm-project/blob/b89211cdd27474469b4e6ca1206c12892a85974b/clang/lib/Sema/SemaDeclAttr.cpp#L3089

`target_version("arch=+c;default")` has already new `HasArch` guard, and it also emit the warning.

`unsupported 'arch=+c;default' in the 'target_version' attribute string; 'target_version' attribute ignored`

Reply:
What about `arch=+c;arch=+f` or `default;default`?

Reply:
The target_version function will now emit a warning when it encounters duplicate attributes.

--- 
I also add the following testcase to cover more invalid version string.
 
1. target_version("arch=+c;arch=+f")
2. target_version("default;default")
3. target_version("priority=1")